sdk/core/core-httpclients/src/xhrHttpClient.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortError } from "@azure/abort-controller";
+import {
+  type HttpClient,
+  type HttpHeaders,
+  type PipelineRequest,
+  type PipelineResponse,
+  type TransferProgressEvent,
+  createHttpHeaders,
+  RestError,
+} from "@azure/core-rest-pipeline";
+
+function isNodeReadableStream(x: unknown): x is NodeJS.ReadableStream {
+  return Boolean(x && typeof (x as NodeJS.ReadableStream)["pipe"] === "function");
+}
+
+function isWebReadableStream(x: unknown): x is ReadableStream {
+  return Boolean(
+    x &&
+      typeof (x as ReadableStream).getReader === "function" &&
+      typeof (x as ReadableStream).tee === "function",
+  );
+}
+
+/**
+ * A HttpClient implementation that uses XMLHttpRequest to send HTTP requests.
+ * @internal
+ */
+class XhrHttpClient implements HttpClient {
+  /**
+   * Makes a request over an underlying transport layer and returns the response.
+   * @param request - The request to be made.
+   */
+  public async sendRequest(request: PipelineRequest): Promise<PipelineResponse> {
+    const url = new URL(request.url);
+    const isInsecure = url.protocol !== "https:";
+
+    if (isInsecure && !request.allowInsecureConnection) {
+      throw new Error(`Cannot connect to ${request.url} while allowInsecureConnection is false.`);
+    }
+
+    const xhr = new XMLHttpRequest();
+
+    if (request.proxySettings) {
+      throw new Error("HTTP proxy is not supported in browser environment");
+    }
+
+    const abortSignal = request.abortSignal;
```

Comment:
Could you do this earlier? Everything so far is synchronous and JS has [run-to-completion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Event_loop#run-to-completion) so I don't _think_ the abort signal can change between 37 and 50 but I may be wrong

Reply:
Won't fix. Exact copy of xisting code.